### PR TITLE
fix(metrics): Add many 'amplitude' event mappings for Settings events

### PR DESCRIPTION
**Pull request by LZoog**
Because:
* Many settings events weren't being properly stored because they weren't mapped to an 'amplitude' event group/name which we currently require for validation and property capturing (uid, flowId etc.)

This commit:
* Combs through our Settings pages for event names and adds ones that were missing, and tweaks a connected-devices event to match regex
* Updates current recovery key events to be in the 'activity' group, since they were not for settings pages
* Adds tests

Fixes [FXA-5762](https://mozilla-hub.atlassian.net/browse/FXA-5762)

---

Reviewer: you might be interested in the comments in that issue and sheets link Wil provided.

Essentially what I did here was search in `fxa-settings` for when we're calling `usePageViewEvent` and `logViewEvent` and checked `amplitude.js` in content-server + tests (**please expand that file in the diff**) to see what we weren't mapping, and also looked at events in [Wil's spreadsheet](https://docs.google.com/spreadsheets/d/17p_8ptV5pX87PSUgtBfqWxrcjRPQQPcdtXSaBWG_ZJc/edit#gid=1557746571) (check the tabs at the bottom). The raw events are only from a day, you'll have to run the query in BigQuery to get latest, if you want to look.

I also went through a few flows in content-server and most of the ones Settings with the Network panel up, checking the `request` POSTing to `/metrics`, just to double check the names were right. In going through content-server, I discovered the `/post_verify/account_recovery/add_recovery_key` URL which you can see after creating an account, which looks like this (without a recovery key - there's a different view with a recovery key)...

~I didn't dig a whole lot here but I saw the events for this page showing up in that spreadsheet under Amplitude events, so users must be seeing it. This was in a "settings" group, which I changed to "registration" since it's under `post_verify`, but I'm not sure if another group is more appropriate, and then added the settings recovery key events.~ See [this comment](https://github.com/mozilla/fxa/pull/14016#issuecomment-1228881506), I used `activity` instead.

**Also**, I realize I probably could have added a couple of these groups to `FUZZY_EVENTS`. However, I found it much easier and more clear _not_ to do this. See [my rant here](https://mozilla.slack.com/archives/CLV3KMZ8B/p1661451602158679?thread_ts=1661451369.450659&cid=CLV3KMZ8B). I'd like us to review how we're doing all this stuff at some point.

And then, I did add a test for all of the new events, but I just went with the most basic tests to verify the mapping. I didn't do data capture validation, which we do sporadically in tests it seems? We're thorough in some event tests, and not in others. I feel like we have _a lot_ of tests in this file already, so 🤷‍♀️ lemme know if you want to see anything in particular.

### CODE REVIEW

**Comment by clouserw** [comment]
I didn't dig a whole lot here but I saw the events for this page showing up in that spreadsheet under Amplitude events, so users must be seeing it. This was in a "settings" group, which I changed to "registration" since it's under post_verify, but I'm not sure if another group is more appropriate, and then added the settings recovery key events.

I don't understand what "under post_verify" means.  I'm not sure these are registration events as you can set a recovery key anytime after you have an account.

**Review by clouserw — APPROVED** [review]
I'm r+, aside from the question of putting all those events into the registration group.

We're going to have a lot of charts to redo after this lands.  I can do that in a couple weeks once this is live.

**Comment by LZoog** [comment]
> I don't understand what "under post_verify" means. I'm not sure these are registration events as you can set a recovery key anytime after you have an account.

@clouserw I mean you can hit them at `/post_verify/account_recovery/*`, like `/post_verify/account_recovery/add_recovery_key`, `/post_verify/account_recovery/verified_recovery_key` etc., and not on Settings. There were only a certain number of [existing groups to choose from](https://github.com/mozilla/fxa/blob/main/packages/fxa-shared/metrics/amplitude.js#L16) - maybe it should be "login" instead? Just not Settings, since Settings recovery key events are separate and I included them in this PR.

File: packages/fxa-content-server/tests/server/amplitude.js
Comment (pdehaan) [inline packages/fxa-content-server/tests/server/amplitude.js:326]: nit: this looks duplicated&hellip; a lot. But not sure how/if we can DRY it slightly.

```js
function createAmplitudeEvent(type='', request=BASIC_REQUEST, data=BASIC_DATA) {
  return amplitude(
    {....getBasicEvent(type)},
    {...request},
    {...data}
  );
}

amplitude(createAmplitudeEvent('screen.settings.change-password'));
```

I guess conversely we could convert the args into an object instead of positional to make it easier to pass just `type` and overridden `BASIC_DATA` values. 🤷 

I can't remember if I saw any instances below of custom `BASIC_REQUEST` or `BASIC_DATA` overrides. But using some sort of utility function might reduce each of these [34] blocks by ~9 lines o' code.

Comment (LZoog) [reply packages/fxa-content-server/tests/server/amplitude.js:326]: 👍 I felt that way about this entire file haha, we don't have anything DRY'd up  here. I tried something similar but not exactly ran into issues so I backed it up a step in DRYness, but I'll try it out, smaller diff is better diff. Thanks!

File: packages/fxa-content-server/server/lib/amplitude.js
Comment (LZoog) [reply packages/fxa-content-server/server/lib/amplitude.js:257]: Yeah we probably do need an issue filed for that, it's mostly just been hanging out in my brain since I'm actively working on something related to this (grooming stuff in [FXA-999](https://mozilla-hub.atlassian.net/browse/FXA-999) / working through the spike in there) and figured as part of wrapping that up I'd remove it but not sure that timing is going to line up perfectly.

**Comment by LZoog** [comment]
@clouserw pushed up that group change to `activity`, I'll plan to merge this Monday morning unless there's any other feedback.

**Comment by clouserw** [comment]
Thanks.  Looks good to me

**Pull request merged**
